backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureExecutor.java

```diff
@@ -0,0 +1,53 @@
+package ai.verta.modeldb.common.futures;
+
+import io.grpc.Context;
+import io.opentracing.Scope;
+import io.opentracing.util.GlobalTracer;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ForkJoinPool;
+import org.springframework.lang.NonNull;
+
+public class FutureExecutor implements Executor {
+  final Executor other;
+
+  FutureExecutor(Executor other) {
+    this.other = other;
+  }
+
+  // Wraps an Executor and make it compatible with grpc's context
+  public static FutureExecutor makeCompatibleExecutor(Executor ex) {
+    return new FutureExecutor(ex);
```

Comment:
I think we should make sure to wrap the executor with otel's `Context.taskWrapping(` executor so that we get tracing-context wrapping automatically, without having to use the open-tracing APIs like we have below.